### PR TITLE
Milestone 6-Jest Libraries:Module 3

### PR DESCRIPTION
**Pull request by Tobinchilongo**
### Math Magician: Test With Jest Libraries:

- Set up React Testing Library.
- Write unit tests for the files operate.js and calculate.js using Jest.
- Use Jest snapshots to test the components.
- Use React Testing Library to simulate user interaction.

### CODE REVIEW

File: src/test/components.test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import Home from '../Pages/Home';
+
+test('testing home', () => {
+  const { container } = render(
+    <MemoryRouter>
+      <Home />
+    </MemoryRouter>,
+  );
```
Comment (medaminedev66) [inline src/test/components.test.js]: - [ ] Following the project requirements, you should test all the React components in your project. 
- Use Jest snapshots to test the components (Calculator, CalculatorBtn, Navigation ...).
- Use React Testing Library to simulate user interaction. Check out this [link ](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/articles/testing_react_with_react_testing_library.md)to learn how to do that.

File: src/test/components.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  </h2>
+  <p>
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  </p>
+  <p>
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  </p>
+</section>
+`);
+});
```
Comment (medaminedev66) [inline src/test/components.test.js]: [OPTIONAL] I would recommend using `react-test-renderer` library and `toMatchSnapshot()` to test the components. Check out this [link ](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/articles/testing_react_with_jest.md)to learn more about that.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi team, :wave:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ] This is a pair programming project, I can see only one developer worked on this project. Please make sure that both of you push commit messages. The reviewer will check the commit history to make sure that everybody has contributed code to the project.

_See the inline comments below for more required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by moise10r — APPROVED** [review]
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi @Tobinchilongo, @awais-amjed** , 

### **Congratulations! 🎉 Project Approved ✅**

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight :1st_place_medal: 
- All the functionalities of the operate.js and calculate.js modules is covered using Jest.✔️
- Usage of Jest snapshots for testing all the components.✔️
- Usage of React Testing Library to simulate user interaction. ✔️  
- All the tests are passing as they should, keep rocking :rocket: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @moise10r when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**